ccl/sqlccl: TestBackupRestoreCrossTableReferences/restore_simple_view failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/bc79f9e2bb0d3d99bb79b99d20aa02e0c33e38fe

Parameters:
```
COCKROACH_PROPOSER_EVALUATED_KV=true
TAGS=
GOFLAGS=
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=219637&tab=buildLog

```

```

Same failure as #14885

```
[02:09:54][Step 2/2]     --- FAIL: TestBackupRestoreCrossTableReferences/restore_simple_view (0.15s)
[02:09:54][Step 2/2]     	backup_test.go:801: pq: use of RESTORE requires an enterprise license. see https://cockroachlabs.com/pricing for details on how to enable enterprise features
```

--- a/pkg/ccl/sqlccl/backup_test.go
+++ b/pkg/ccl/sqlccl/backup_test.go
@@ -105,8 +105,23 @@ func bankSplitStmt(numAccounts int, numRanges int) string {
 	return stmt.String()
 }
 
-func enableEnterprise(db *sqlutils.SQLRunner) {
-	db.Exec("SET CLUSTER SETTING enterprise.enabled = true")
+func backupSQLRunner(t testing.TB, tc *testcluster.TestCluster) *sqlutils.SQLRunner {
+	runner := sqlutils.MakeSQLRunner(t, tc.Conns[0])
+	runner.Exec("SET CLUSTER SETTING enterprise.enabled = true")
+
+	show := "SHOW CLUSTER SETTING enterprise.enabled"
+	for _, conn := range tc.Conns {
+		testutils.SucceedsSoon(t, func() error {
+			var enabled string
+			sqlutils.MakeSQLRunner(t, conn).QueryRow(show).Scan(&enabled)
+			if enabled != "true" {
+				return errors.New("enterprise.enabled is not true")
+			}
+			return nil
+		})
+	}
+
+	return runner
 }
 
 func backupRestoreTestSetupWithParams(
@@ -133,8 +148,7 @@ func backupRestoreTestSetupWithParams(
 		}
 	}
 
-	sqlDB = sqlutils.MakeSQLRunner(t, tc.Conns[0])
-	enableEnterprise(sqlDB)
+	sqlDB = backupSQLRunner(t, tc)
 
 	sqlDB.Exec(bankCreateDatabase)
 
@@ -276,8 +290,7 @@ func backupAndRestore(
 	{
 		tcRestore := testcluster.StartTestCluster(t, multiNode, base.TestClusterArgs{})
 		defer tcRestore.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tcRestore.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tcRestore)
 
 		// Create some other descriptors to change up IDs
 		sqlDBRestore.Exec(`CREATE DATABASE other`)
@@ -516,8 +529,7 @@ func TestBackupRestoreInterleaved(t *testing.T) {
 	t.Run("all tables in interleave hierarchy", func(t *testing.T) {
 		tcRestore := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tcRestore.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tcRestore.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tcRestore)
 		sqlDBRestore.Exec(bankCreateDatabase)
 
 		sqlDBRestore.Exec(`RESTORE bench.* FROM $1`, dir)
@@ -544,8 +556,7 @@ func TestBackupRestoreInterleaved(t *testing.T) {
 	t.Run("interleaved table without parent", func(t *testing.T) {
 		tcRestore := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tcRestore.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tcRestore.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tcRestore)
 		sqlDBRestore.Exec(bankCreateDatabase)
 
 		_, err := sqlDBRestore.DB.Exec(`RESTORE TABLE bench.i0 FROM $1`, dir)
@@ -557,8 +568,7 @@ func TestBackupRestoreInterleaved(t *testing.T) {
 	t.Run("interleaved table without child", func(t *testing.T) {
 		tcRestore := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tcRestore.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tcRestore.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tcRestore)
 		sqlDBRestore.Exec(bankCreateDatabase)
 
 		_, err := sqlDBRestore.DB.Exec(`RESTORE TABLE bench.bank FROM $1`, dir)
@@ -651,8 +661,8 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore everything to new cluster", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
+
 		db.Exec(createStore)
 		db.Exec(`RESTORE store.* FROM $1`, dir)
 		// Restore's Validate checks all the tables point to each other correctly.
@@ -694,8 +704,7 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore customers to new cluster", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
 		db.Exec(createStore)
 		db.Exec(`RESTORE store.customers, store.orders FROM $1`, dir)
 		// Restore's Validate checks all the tables point to each other correctly.
@@ -714,8 +723,7 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore orders to new cluster", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
 		db.Exec(createStore)
 
 		// FK validation of self-FK is preserved.
@@ -738,8 +746,7 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore receipts to new cluster", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
 		db.Exec(createStore)
 		db.Exec(`RESTORE store.receipts FROM $1 WITH OPTIONS ('skip_missing_foreign_keys')`, dir)
 		// Restore's Validate checks all the tables point to each other correctly.
@@ -758,8 +765,7 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore receipts and customers to new cluster", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
 		db.Exec(createStore)
 		db.Exec(`RESTORE store.receipts, store.customers FROM $1 WITH OPTIONS ('skip_missing_foreign_keys')`, dir)
 		// Restore's Validate checks all the tables point to each other correctly.
@@ -792,8 +798,7 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore simple view", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
 		db.Exec(createStore)
 		if _, err := db.DB.Exec(`RESTORE store.early_customers FROM $1`, dir); !testutils.IsError(err,
 			`cannot restore "early_customers" without restoring referenced table`,
@@ -823,8 +828,7 @@ func TestBackupRestoreCrossTableReferences(t *testing.T) {
 	t.Run("restore multi-table view", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		db := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(db)
+		db := backupSQLRunner(t, tc)
 		db.Exec(createStore)
 		db.Exec(createStoreStats)
 
@@ -946,8 +950,8 @@ func TestBackupRestoreIncremental(t *testing.T) {
 	{
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tc)
+
 		sqlDBRestore.Exec(`CREATE DATABASE bench`)
 
 		for i := len(backupDirs); i > 0; i-- {
@@ -1352,8 +1356,7 @@ func TestRestoredPrivileges(t *testing.T) {
 	t.Run("into fresh db", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tc)
 		sqlDBRestore.Exec(`CREATE DATABASE bench`)
 		sqlDBRestore.Exec(`RESTORE bench.bank FROM $1`, dir)
 		sqlDBRestore.CheckQueryResults(`SHOW GRANTS ON bench.bank`, rootOnly)
@@ -1362,8 +1365,7 @@ func TestRestoredPrivileges(t *testing.T) {
 	t.Run("into db with added grants", func(t *testing.T) {
 		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{})
 		defer tc.Stopper().Stop()
-		sqlDBRestore := sqlutils.MakeSQLRunner(t, tc.Conns[0])
-		enableEnterprise(sqlDBRestore)
+		sqlDBRestore := backupSQLRunner(t, tc)
 		sqlDBRestore.Exec(`CREATE DATABASE bench`)
 		sqlDBRestore.Exec(`CREATE USER someone`)
 		sqlDBRestore.Exec(`GRANT SELECT, INSERT, UPDATE, DELETE ON DATABASE bench TO someone`)

